src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobUtils.scala

```diff
@@ -103,41 +105,49 @@ object JobUtils {
   }
 
   def makeScheduleStream(job: ScheduleBasedJob, dateTime: DateTime) = {
-    Iso8601Expressions.parse(job.schedule, job.scheduleTimeZone) match {
-      case Some((_, scheduledTime, _)) =>
-        if (scheduledTime.plus(job.epsilon).isBefore(dateTime)) {
+    job.schedule match {
+      case schedule: ISO8601Schedule => {
+        if(schedule.start.plus(job.epsilon).isBefore(dateTime)) {
           skipForward(job, dateTime)
         } else {
           Some(new ScheduleStream(job.schedule, job.name, job.scheduleTimeZone))
         }
-      case None =>
+      }
+      //TODO: add cron support
+      case _ =>
         None
     }
   }
 
   def skipForward(job: ScheduleBasedJob, dateTime: DateTime): Option[ScheduleStream] = {
-    Iso8601Expressions.parse(job.schedule, job.scheduleTimeZone) match {
-      case Some((rec, start, per)) =>
-        val skip = calculateSkips(dateTime, start, per)
-        if (rec == -1) {
-          val nStart = start.plus(per.multipliedBy(skip))
+    job.schedule match {
+      case schedule: ISO8601Schedule => {
+        val skip = calculateSkips(dateTime, schedule.start, schedule.period)
```

Comment:
er, I instead created a function to transform a schedule, and then made a factory for creating streams for a given date (so that we don't have to calculate skips here)